File: rental_manager_server/src/main/kotlin/pl/monochrome/manager/model/database/RentalObject.kt
package pl.monochrome.manager.model.database

import org.hibernate.annotations.Type
import org.hibernate.annotations.TypeDef
import pl.monochrome.manager.model.enums.PostgresqlEnumType
import pl.monochrome.manager.model.enums.ObjectType
import javax.persistence.*

@Entity
@TypeDef(
    name = "pgsql_enum",
    typeClass = PostgresqlEnumType::class
)
@Table(name = "Rental_objects")
data class RentalObject(

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int,

    @Column
    val maxGuests: Int,

    @Column
    val description: String,

    @Column
    val name: String,

    @Column
    @Enumerated(EnumType.STRING)
    @Type(type = "pgsql_enum")
    val type: ObjectType,

    @Column
    val allowedAnimals: Boolean,

    @ManyToOne
    val user: User

)

File: rental_manager_app/lib/model/rental_object.dart
import 'package:rental_manager_app/model/user.dart';

class RentalObject {
  int id;
  int maxGuests;
  String description;
  String name;
  String type;
  bool allowedAnimals;
  User user;

  RentalObject(
      {this.id,
        this.maxGuests,
        this.description,
        this.name,
        this.type,
        this.allowedAnimals,
        this.user});

  RentalObject.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    maxGuests = json['maxGuests'];
    description = json['description'];
    name = json['name'];
    type = json['type'];
    allowedAnimals = json['allowedAnimals'];
    user = json['user'] != null ? new User.fromJson(json['user']) : null;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['maxGuests'] = this.maxGuests;
    data['description'] = this.description;
    data['name'] = this.name;
    data['type'] = this.type;
    data['allowedAnimals'] = this.allowedAnimals;
    if (this.user != null) {
      data['userId'] = this.user.id;
    }
    return data;
  }
}